multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessMtaArchiveStep.java

```diff
@@ -1,78 +1,90 @@
         MtaArchiveElements mtaArchiveElements = new MtaArchiveElements();
         addMtaArchiveModulesInMtaArchiveElements(context, helper, mtaArchiveElements);
         addMtaRequiredDependenciesInMtaArchiveElements(helper, mtaArchiveElements);
         addMtaArchiveResourcesInMtaArchiveElements(helper, mtaArchiveElements);
         context.setVariable(Variables.MTA_ARCHIVE_ELEMENTS, mtaArchiveElements);
+        DeploymentDescriptor deploymentDescriptor = extractDeploymentDescriptor(context, appArchiveId, archiveEntriesWithStreamPositions);
+        context.setVariable(Variables.DEPLOYMENT_DESCRIPTOR, deploymentDescriptor);
     }
 
-    private DeploymentDescriptor extractDeploymentDescriptor(InputStream appArchiveStream) {
-        String descriptorString = ArchiveHandler.getDescriptor(appArchiveStream, configuration.getMaxMtaDescriptorSize());
-        getStepLogger().debug("MTA Descriptor length: {0}", descriptorString.length());
-        DescriptorParserFacade descriptorParserFacade = descriptorParserFactory.getInstance();
-        return descriptorParserFacade.parseDeploymentDescriptor(descriptorString);
-    }
-
-    private MtaArchiveHelper createMtaArchiveHelperFromManifest(InputStream appArchiveStream) {
-        Manifest manifest = ArchiveHandler.getManifest(appArchiveStream, configuration.getMaxManifestSize());
-        MtaArchiveHelper helper = getHelper(manifest);
-        helper.init();
-        return helper;
+    private MtaArchiveHelper createMtaArchiveHelperFromManifest(ProcessContext context, String appArchiveId,
+                                                                List<ArchiveEntryWithStreamPositions> archiveEntriesWithStreamPositions) {
+        ArchiveEntryWithStreamPositions mtaManifestEntry = ArchiveEntryExtractorUtil.findEntry(ArchiveHandler.MTA_MANIFEST_NAME,
+                                                                                               archiveEntriesWithStreamPositions);
+        byte[] inflatedManifestFile = archiveEntryExtractor.readFullEntry(ImmutableFileEntryProperties.builder()
+                                                                                                      .guid(appArchiveId)
+                                                                                                      .name(mtaManifestEntry.getName())
+                                                                                                      .spaceGuid(context.getRequiredVariable(Variables.SPACE_GUID))
+                                                                                                      .maxFileSizeInBytes(configuration.getMaxMtaDescriptorSize())
+                                                                                                      .build(),
+                                                                          mtaManifestEntry);
```

Comment:
Extract in a separate method that gets the inflated manifest file

Reply:
done